src/System.Private.ServiceModel/tests/Scenarios/SelfHostWcfService/TestResources/EndpointResource.cs

```diff
@@ -32,10 +32,22 @@ public object Put()
                     {
                         host = new ServiceHost(typeof(ServiceType));
                         // URI assumes localhost but that instead should be passed in
+                        // For Kerberos, localhost cannot be used as localhost will fall back to NTLM
+                        // this is a workaround until the URI is passed in
+                        Binding binding = GetBinding();
+                        string baseAddress = "localhost";
+                        if (binding is BasicHttpBinding)
```

Comment:
+1  having and overriable method that builds the service host would be ideal and Put just calls it and have the Httpkerb method implement and know about these types. Checking for binding and special casing in the base class makes this fragile in the long run.